base/src/main/java/org/aya/terck/CallGraph.java

```diff
@@ -20,72 +22,126 @@
  * @see CallMatrix
  */
 public record CallGraph<T, P>(
```

Comment:
It might be a good idea to move this to `tools`?

Reply:
Yes this is planned

Reply:
So other projects (like guest, anqur) can have termination check for free!